src/proxy/reconnect.rs

```diff
@@ -169,9 +174,21 @@ where
     type Future = <Reconnect<M, T> as svc::Service<Req>>::Future;
 
     fn poll_ready(&mut self) -> Poll<(), Self::Error> {
+        fn apply_jitter(duration: Duration, max_jitter: f64) -> Duration {
+            let jitter = rand::random::<f64>() * max_jitter;
+            let secs = (duration.as_secs() as f64) * jitter;
+            let nanos = (duration.subsec_nanos() as f64) * jitter;
+            let millis = (secs * 1000f64) + (nanos / 1000000f64);
+            duration + Duration::from_millis(millis as u64)
+        }
+
+        fn next_backoff(failed_attempts: u32, min: &Duration, max: &Duration) -> Duration {
```

Comment:
What do you think about moving the logic from `next_backoff` & `apply_jitter` into a method on `Backoff`?

For example, something like the following:

```rust
impl Backoff {
  fn for_failures<R: rand::Rng>(&self, failures: u32, rng: R) -> Duration {
    match self {
      Backoff::None => Duration::from_secs(0),
      Backoff::Exponential { max, min, jitter } => { ... }
    }
  }
}
```

Then, we can write unit tests on the `Backoff` type independently of the reconnect logic!